tools/screenshots.sh

```diff
@@ -0,0 +1,748 @@
+    echo ""
+    exit 1
+}
+
+# Show Helpers
+function show_error_message() {
+    message=$1
+    echo "$OUTPUT_RED$message$OUTPUT_NORM"
+    echo $message >> $LOG_FILE
+}
+
+
+function show_warning_message() {
+    message=$1
+    echo "$OUTPUT_YELLOW$message$OUTPUT_NORM"
+    echo $message >> $LOG_FILE
+}
+
+function show_ok_message() {
+    message=$1
+    echo "$OUTPUT_GREEN$message$OUTPUT_NORM" 
+    echo $message >> $LOG_FILE
+}
+
+function show_title_message() {
+    message=$1
+    echo "$OUTPUT_BOLD$message$OUTPUT_NORM" 
+    echo $message >> $LOG_FILE
+}
+
+function show_message() {
+    echo "$1" | tee -a $LOG_FILE
+}
+
+# Appends an init line to the log
+function start_log() {
+    dateTime=`date "+%d-%m-%Y - %H:%M:%S"`
+    echo "$exeName started at $dateTime" >> $LOG_FILE
+}
+
+# Appends a closing line to the log
+function stop_log() {
+    dateTime=`date "+%d-%m-%Y - %H:%M:%S"`
+    echo "$exeName terminated at $dateTime" >> $LOG_FILE
+    echo "" >> $LOG_FILE
+    echo "Log location: $LOG_FILE"
+}
+
+# Writes an error message and exits
+function stop_on_error() {
+    show_error_message "Operation failed. Aborting."
+    show_error_message "See log for further details."
+    stop_log
+    exit 1
+}
+
+# Shows the current configuration
+function show_config() {
+  show_title_message "Configuration:"
+  show_message "Device: $RUN_DEV"
+  show_message "Emulator: $AVD"
+  show_message "Package: $APK ($PKG)"
+}
+
+# Checks and setups the working dir
+function require_dirs {
+  if [ ! -d "$WORKING_DIR" ]; then
+    show_message Creating working directory...
+    mkdir "$WORKING_DIR" >> $LOG_FILE 2>&1 || stop_on_error
+    show_message Done
+  fi 
+}
+
+# Checks and setups the working dir for taking the screenshots
+function require_image_dirs() {
+  compDir=$1/$2/$3  
+  require_dirs
+
+  if [ ! -d "$compDir" ]; then
+    show_message "Creating screenshot directory..."
+    mkdir -p "$compDir" >> $LOG_FILE 2>&1 || stop_on_error
+    show_message "Done"
+  fi 
+}
+
+# Checks the availability of the promo font
+function require_font {
+  if [ ! -f "$FONT_FILE" ]; then
+    show_message "Font file is missing."
+    require_dirs
+
+    show_message "Downloading..."
+    command -v wget >/dev/null 2>&1 || { show_error_message "wget command not installed. Please, install it!"; stop_on_error; }
+    wget $FONT_ZIP_URL -O "$WORKING_DIR/noto.zip" >> $LOG_FILE 2>&1 || stop_on_error
+    show_message "Done"
+
+    show_message "Unzipping..."
+    unzip "$WORKING_DIR/noto.zip" -d "$WORKING_DIR/$FONT_DIR/" >> $LOG_FILE 2>&1 || stop_on_error
+    show_message "Done"
+  fi
+}
+
+# Checks for ImageMagick
+function require_imagemagick {
+  which magick &>/dev/null
+  immissing=$?
+
+  if [ $immissing = 1 ]; then
+    show_message "Installing ImageMagick..."
+    exec brew install imagemagick
+  fi
+}
+
+# Setups the local sdk and configure the environment
+function require_sdk() {
```

Comment:
I would really drop this function, I expect people doing the screenshots to have the Android SDK installed. Installing it for them seems like an overkill, especially since it will be installed under the wp repo.

We should just have a check and show an error if the sdk is not installed. We could also add the following to the config file:

```
ANDROID_SDK_DIR=FIXME
PATH=$PATH:$ANDROID_SDK_DIR/tools/bin/
```

Reply:
require_sdk is only called by the "setup" if the required tools (adb, emu, etc.) are not found, so if people already has the sdk installed, it would have no effect.
Anyway, it surely makes no sense to maintain code that is not used, so ok for dropping it. 

At this point, I wonder if the same applies for the creation of the default emulator. 
Do you think we should drop it too and with it remove the "setup" option?